java-cfenv-boot-pivotal-sso/src/main/java/io/pivotal/cfenv/boot/sso/CfSingleSignOnProcessor.java

```diff
@@ -59,6 +61,9 @@ public void process(CfCredentials cfCredentials, Map<String, Object> properties)
         properties.put(SSO_SERVICE, authDomain);
         properties.put(SPRING_SECURITY_CLIENT + ".provider." + PROVIDER_ID + ".issuer-uri", issuer + "/oauth/token");
         properties.put(SPRING_SECURITY_CLIENT + ".provider." + PROVIDER_ID + ".authorization-uri", authDomain + "/oauth/authorize");
+        if(SpringSecurityDetector.isSpringResourceServerPresent()) {
+            properties.put("spring.security.oauth2.resourceserver.jwt.issuer-uri", issuer + "/oauth/token");
```

Comment:
Are you sure that it's the `/oauth/token` endpoint that's needed here?

Reply:
I think the relevant doc is here: https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-security
Specifically, section 10.3.2

Reply:
If it's working like this, maybe it just needs the hostname and doesn't care about the path.

Reply:
No. It has to be exactly like this otherwise issuer validation fails. See what UAA sets as issuer:
`https://login.run.pcfone.io/.well-known/openid-configuration`

Reply:
UAA has a quirk where issuer actually doesn't match the SSO well-known endpoint. When the authorization endpoint is set it automatically bootstraps it from OIDC well-known configuration, but the issuer needs to be overridden to pass validation.

Reply:
Ok, fair enough!